src/library/baseplaylistfeature.cpp

```diff
@@ -156,6 +156,7 @@ void BasePlaylistFeature::activateChild(const QModelIndex& index) {
     //qDebug() << "BasePlaylistFeature::activateChild()" << index;
     int playlistId = playlistIdFromIndex(index);
     if (playlistId != -1) {
+        emit saveModelState();
```

Comment:
I think we can get rid of the library signal roundtrip if we move emit saveModelState(); into m_pPlaylistTableModel->setTableModel(playlistId); and connect it directly to WLibraryTableView using model() and qobject_cast to TrackModel

Reply:
I tried this but this does not work and I tried it first. Many features like use the same model or do stuff in the background and save and restore must be done at certain points. For example, the crate model changes the crateId for the same model and calls to active itself. At that point the id variable is already changed but the model still has the old data.

Reply:
OK.